src/engine/strat_engine/liminal/liminal.rs

```diff
@@ -233,10 +276,229 @@ impl LiminalDevices {
+                self.handle_stopped_pool(pool_uuid, ds);
+                return Err(err);
+            }
         };
 
-        self.try_setup_pool(pools, pool_uuid, stopped_pool)
-            .map(|(name, pool)| (name, pool_uuid, pool, uuids))
+        let res = load_stratis_metadata(pool_uuid, stratis_infos_ref(&infos));
+        let (infos, bdas) = split_stratis_infos(infos);
+
+        match start_pool_failure(pools, pool_uuid, luks_info, &infos, bdas, res) {
+            Ok((name, pool)) => {
+                self.uuid_lookup = self
+                    .uuid_lookup
+                    .drain()
+                    .filter(|(_, (p, _))| *p != pool_uuid)
+                    .collect();
+                self.name_to_uuid = self
+                    .name_to_uuid
+                    .drain()
+                    .filter_map(|(n, mut maybe_conflict)| {
+                        if maybe_conflict.remove(&pool_uuid) {
+                            None
+                        } else {
+                            Some((n, maybe_conflict))
+                        }
+                    })
+                    .collect();
+                info!(
+                    "Pool with name \"{}\" and UUID \"{}\" set up",
+                    name, pool_uuid
+                );
+                Ok((name, pool_uuid, pool, uuids))
+            }
+            Err((err, bdas)) => {
+                info!("Attempt to set up pool failed, but it may be possible to set up the pool later, if the situation changes: {}", err);
+                let device_set = reconstruct_stratis_infos(infos, bdas);
+                self.handle_stopped_pool(pool_uuid, device_set);
+                Err(err)
+            }
+        }
+    }
+
+    /// Start a pool, create the devicemapper devices, and return the fully constructed
+    /// pool.
+    pub fn start_pool_new(
+        &mut self,
+        pools: &Table<PoolUuid, AnyPool>,
+        pool_uuid: PoolUuid,
+        unlock_method: Option<UnlockMethod>,
+        passphrase_fd: Option<RawFd>,
+    ) -> StratisResult<(Name, PoolUuid, AnyPool, Vec<DevUuid>)> {
+        fn start_pool_failure(
+            pools: &Table<PoolUuid, AnyPool>,
+            pool_uuid: PoolUuid,
+            infos: &HashMap<DevUuid, LStratisDevInfo>,
+            bdas: HashMap<DevUuid, BDA>,
+            meta_res: StratisResult<(DateTime<Utc>, PoolSave)>,
+            unlock_method: Option<UnlockMethod>,
+            passphrase_fd: Option<RawFd>,
+        ) -> BDARecordResult<(Name, AnyPool)> {
+            let (timestamp, metadata) = match meta_res {
+                Ok(o) => o,
+                Err(e) => return Err((e, bdas)),
+            };
+
+            let passphrase = match (
+                metadata.features.contains(PoolFeatures::Encryption),
+                unlock_method,
+                passphrase_fd,
+            ) {
+                (false, _, Some(_)) => {
```

Comment:
Could you convert lines 361 to lines 368 (two matches) to the following two matches instead:

```
(false, None, None) => { None },
(false, _, _) => {
    return Err((StratisError::Msg(format!("Metadata reported that encryption was not enabled but some unlock method was provided.")));
}
```
}